### PR TITLE
nameof_enum_or to fallback to numeric representation

### PR DESCRIPTION
**Pull request by avlec**
Implements #41
macro and implementation, and small unit test to get the ball rolling on discussion

### CODE REVIEW

File: include/nameof.hpp [hunk 1]
```diff
@@ -1002,11 +1001,29 @@ template <typename E>
   return {}; // Value out of range.
 }
 
+
+template <typename E, typename D = std::decay_t<E>, typename U = std::underlying_type_t<D>>
+[[nodiscard]] constexpr auto nameof_enum_or(E value) noexcept -> detail::enable_if_enum_t<E, string> {
```
Comment (schaumb) [inline include/nameof.hpp:1006]: string return value is not constexpr. Probably here need string_view

File: include/nameof.hpp [hunk 2]
```diff
@@ -1123,6 +1136,9 @@ template <auto V>
 // Obtains name of enum variable.
 #define NAMEOF_ENUM(...) ::nameof::nameof_enum<::std::decay_t<decltype(__VA_ARGS__)>>(__VA_ARGS__)
 
+// Obtains name of enum variable or stringified number if out of range.
+#define NAMEOF_ENUM_OR(...) ::nameof::nameof_enum_or<::std::decay_t<decltype(__VA_ARGS__)>>(__VA_ARGS)
+
```
Comment (schaumb) [inline include/nameof.hpp:1141]: Missing 2 underscore from 2nd `__VA_ARGS__` macro

File: include/nameof.hpp (hunk 1 above)
Comment (avlec) [reply include/nameof.hpp:1006]: I thought I had removed the constexpr, oh well it's been closed anyways

Comment (Neargye) [reply include/nameof.hpp:1006]: I will make a separate PR to add a customization option

**Pull request closed without merging**